onnxruntime/core/optimizer/qdq_transformer/selectors_actions/shared/utils.h

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <string>
+#include "core/graph/basic_types.h"
+#include "core/providers/nnapi/nnapi_builtin/nnapi_lib/NeuralNetworksTypes.h"
+#include "core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.h"
+#include "core/optimizer/selectors_actions/helpers.h"
+
+namespace onnxruntime {
+
+class GraphViewer;
+class Node;
+
+namespace QDQ {
+
+struct OpVersionsAndSelector {
+  using OpVersionsMap = std::unordered_map<std::string, std::vector<ONNX_NAMESPACE::OperatorSetVersion>>;
+
+  OpVersionsAndSelector(const OpVersionsMap& ops_and_versions_in,
+                        std::unique_ptr<BaseSelector> selector_in)
+      : op_versions_map{ops_and_versions_in},
+        selector{std::move(selector_in)} {}
+
+  OpVersionsMap op_versions_map;
+  std::unique_ptr<BaseSelector> selector;
+
+  ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE(OpVersionsAndSelector);
+};
+
+class Selectors {
+ public:
+  Selectors() = default;
+
+  void RegisterSelector(const OpVersionsAndSelector::OpVersionsMap& ops_and_versions_in,
+                        std::unique_ptr<BaseSelector> selector_in);
+
+  const std::unordered_set<std::unique_ptr<OpVersionsAndSelector>>& SelectorsSet() const {
+    return selectors_set_;
+  }
+
+  ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE(Selectors);
+
+ private:
+  std::unordered_set<std::unique_ptr<OpVersionsAndSelector>> selectors_set_;
+};
+
+class SelectorManager {
+ public:
+  SelectorManager() = default;
+
+  void Initialize();
+
+  std::vector<NodeGroup> GetQDQSelections(const GraphViewer& graph_viewer) const;
+
```

Comment:
Some documentation for non-trivial public methods would be good. 

e.g. Initialize is self-explanatory. GetQDQSelections could provide a quick overview of what happens. Imagine you're a new developer trying to understand what the class does. Without that info you're forced to read the entire implementation of the method to know this. That's both time consuming and potentially error prone if something in the implementation is misinterpreted.